### ISSUE TITLE
'cache_keys.normalize_url_params' is not comptabile with all params types accepted by 'aiohttp' (that uses 'yarl' package under the hood)

### ISSUE THREAD

**Issue by AIGeneratedUsername** [JWCook/aiohttp-client-cache#105]
### The problem
`cache_keys.normalize_url_params` is not comptabile with all `params` variable types accepted by '**aiohttp**' (that uses '**yarl**' package under the hood).

**aiohttp** validates `params` via `yarl._url.URL._get_str_query`.

`yarl._url.URL._get_str_query` accepts:
- dict of strings
-  dict of floats
- iterable of pairs (I think in Python it is named `Mapping`, but correct me if I am wrong). Example: tuple of tuples `(("foo", "bar), ("spam", "eggs"))`). 

### Expected behavior

Accept the same `params` types as **aiohttp**.

### Steps to reproduce the behavior
```python
import asyncio

from aiohttp_client_cache import CachedSession, SQLiteBackend

cache = SQLiteBackend(
    "stream_test",
    urls_expire_after={
        "httpbin.org": 0,
    },
)


async def check_json_params():
    async with CachedSession(cache=cache) as session:
        # See `yarl._url.URL._get_str_query` for all accepted types.
        response = await session.post("https://httpbin.org/json", params="foobar")
        print("From cache:", response.from_cache)

async def main():
    await cache.clear()
    await check_json_params()
    await check_json_params()


if __name__ == "__main__":
    asyncio.run(main())
```

### Workarounds

No.

The `normalize_url_params` must be fixed to accept the same types as `yarl._url.URL._get_str_query` accepts.

### Environment
- aiohttp-client-cache version: 0.5.2
- Python version: `3.9`
- Platform: Lubuntu 20.04

**Comment by AIGeneratedUsername** [comment]
Here is a test sample for you. You can use it to make sure that handle all `params` types. It is a sample code: copy possible input types and adapt to your project.

```python
fake = Faker()


# Note: it is not a best practice to make calls in `parametrize`, but there is no any clean way to pass fixtures into it.
# For details see issue: https://github.com/pytest-dev/pytest/issues/349
@pytest.mark.parametrize(
    "params, raises_error",
    [
        ({fake.pystr(): lambda x: x}, True),  # Dict with Callables
        ({fake.pystr(): fake.pydict()}, True),  # Dict of Dicts
        (fake.binary(length=16), True),  # bytes
        ({fake.pystr(): fake.boolean()}, True),  # Dict of bools
        (fake.pyint(), True),  # int
        (fake.pyfloat(), True),  # float
        ({fake.pystr(): None}, True),  # Dict of None
        (fake.pydict(value_types=(str, bool, float)), True),  # Dict with mixed values (including forbidden)
        (
            tuple([fake.pytuple(nb_elements=2, variable_nb_elements=False, value_types=(str, str)) for _ in range(fake.pyint(max_value=50))]),
            False,
        ),  # tuple of tuples
        ({fake.pystr(): fake.pyfloat()}, False),  # Dict of floats
        ({fake.pystr(): fake.pystr()}, False),  # Dict of strings
        (fake.pystr(), False),  # string
    ],
)
@pytest.mark.asyncio()
async def test_request_params(params: Any, raises_error: bool) -> None:
    # Write your test here...
```

**Comment by JWCook** [comment]
Thanks for reporting that! I'll get that fixed soon.

**Comment by JWCook** [comment]
Fixed in `0.6.0.dev1`. Thanks for the complete example and tests!
